### ISSUE TITLE
Tutorial

### ISSUE THREAD

**Issue by MiroQurcika** [markusfisch/ShaderEditor#48]
I think we need small tutorial to pick up basic things

**Comment by MiroQurcika** [comment]
We just need to learn to use this program 😕

**Comment by markusfisch** [comment]
Well, yes, a tutorial would be good, of course. On the other hand, GLSL might be too broad a topic for an in-app tutorial.

In the meantime, please check out the links from the [README](https://github.com/markusfisch/ShaderEditor#where-can-i-learn-more-about-shaders).

**Comment by VP-** [comment]
No, not GLSL. There are enough tutorials about that on the internet already. What is missing is a tutorial (or any documentation, really) on the app itself.

For example: how do I use the camera? That's not part of GLSL, that's (probably) a `uniform sampler2D` automatically provided by the app, as soon as I declare it in my shader. But there is not even a list of **names** of these `uniform`s, much less any information on how to use them.

**Comment by markusfisch** [comment]
Well, actually there's is a list of all available uniforms. In the overflow menu, pick "Add uniform/texture":

![screencap-20201124192452](https://user-images.githubusercontent.com/485246/100136214-04c0ad00-2e8b-11eb-9c63-ce09a623a6b9.png)

Also, there's a camera sample in "Load sample" named "Back camera" 😉

**Comment by VP-** [comment]
Awesome! That's exactly what I was missing. Could you add a reference to these two menu items to the FAQ?

**Comment by thornySoap** [comment]
In the README it says

> Import and use arbitrary textures

But I have troubles finding out how to sample from an specific image. Is it even possible or am I misunderstanding something? Either way maybe some sort of little guidance in the app would be nice

**Comment by markusfisch** [comment]
In a nutshell:

How to draw textures:
Menu > Load samples > Texture 

How to load a texture:
Menu > Add uniform/texture > swipe right > tap on the circle button in the bottom right corner 

Where Menu is the three dots in the upper right corner.

If you already imported an image and wonder how to actually use it: you will need to add a uniform with the name of the texture. Check out the Texture sample on how to do this.

**Comment by thornySoap** [comment]
Oohh I see, you can swipe on that screen. Thank you very much. I've seen the noise example but couldn't figure out how to import custom ones.

Maybe it would be nice if there was a tab bar at the top of the "Add uniform/texture" screen to indicate there is more than the first page?

**Comment by markusfisch** [comment]
Actually, there is a tab bar - it's just not visible due to a [bug](https://github.com/markusfisch/ShaderEditor/issues/185) in the lastest release 🙈 Will be fixed in the next version!

**Comment by asorc** [comment]
Ok, now I know finally how to do It. I thought the texture noise was not an image but a texture created from an algorithm built in the app itself, since usually, applications and programa have built in functions to generate different kind of noises. 

Thanks for the information. It's not obvious that you have to swipe right. Thank you. Thanks. 

Cheers.
